napari/utils/colormaps/standardize_color.py

```diff
@@ -0,0 +1,331 @@
+"""This module contains functions that 'standardize' the color handling
+of napari layers by supplying functions that are able to convert most
+color representation the user had in mind into a single representation -
+a numpy Nx4 array of float32 values - that is used across the codebase.
+
+The main function of the module is "transform_color", which might call
+a cascade of other, private, function in the module to do the hard work
+of converting the input.
+
+In general, we try to catch invalid color representations, warn the users
+of their misbehaving and return a default white color array, since it seems
+unreasonable to crash the entire napari session to mis-represented colors.
+"""
+
+import types
+from typing import Iterable, Union, Dict, Any, Callable
+import functools
+import warnings
+
+import numpy as np
+from vispy.color import Color, ColorArray, get_color_dict, get_color_names
+from vispy.color.color_array import _string_to_rgb
+
+
+def transform_color(colors: Any) -> np.ndarray:
+    """Receives the user-given colors and transforms them into an array of
```

Comment:
Follow PEP257: single line, present imperative, followed by a blank line, followed if desired by a longer paragraph. e.g.

```suggestion
    """Transform provided color(s) to an Nx4 array of RGBA np.float32 values.
    
    N is the number of given colors. The function is designed to parse all valid
    ...
```